src/mlpack/core/data/stb_image.h

```diff
@@ -0,0 +1,7988 @@
+/* stb_image - v2.30 - public domain image loader - http://nothings.org/stb
```

Comment:
Might be better to put this in `core/stb/` so it is clear to someone browsing that this is a bundled dependency.